.ci/CI/src/Bootloader.jl

```diff
@@ -0,0 +1,380 @@
+    )
+    job_yaml["image"] = "julia:$(version)"
+
+    script = [
+        "apt update && apt install -y git",
+        "git clone --depth 1 -b $(git_ci_tools_branch) $(git_ci_tools_url) /tmp/integration_test_tools/",
+    ]
+
+    if target_branch == "main"
+        push!(
+            script,
+            "julia --project=. /tmp/integration_test_tools/.ci/CI/src/SetupDevEnv.jl \${CI_PROJECT_DIR}/Project.toml NO_MESSAGE",
+        )
+    else
+        push!(
+            script,
+            "julia --project=. /tmp/integration_test_tools/.ci/CI/src/SetupDevEnv.jl \${CI_PROJECT_DIR}/Project.toml",
+        )
+    end
+
+    script = vcat(
+        script,
+        [
+            "julia --project=. -e 'import Pkg; Pkg.instantiate()'",
+            "julia --project=. -e 'import Pkg; Pkg.test(; coverage = true)'",
+        ],
+    )
+    job_yaml["script"] = script
+
+    job_yaml["interruptible"] = true
+    job_yaml["tags"] = ["cpuonly"]
+
+    return job_yaml
+end
+
+function get_nighlty_unit_test(
+    target_branch::String,
+    dev_package_name::AbstractString,
+    dev_package_version::AbstractString,
+    dev_package_path::AbstractString,
+    git_ci_tools_url::String,
+    git_ci_tools_branch::String,
+)
+    job_yaml = get_normal_unit_test(
+        "1",
+        target_branch,
+        dev_package_name,
+        dev_package_version,
+        dev_package_path,
+        git_ci_tools_url,
+        git_ci_tools_branch,
+    )
+    job_yaml["image"] = "debian:bookworm-slim"
+
+    if !haskey(job_yaml, "variables")
+        job_yaml["variables"] = Dict()
+    end
+    job_yaml["variables"]["JULIA_DONWLOAD"] = "/julia/download"
+    job_yaml["variables"]["JULIA_EXTRACT"] = "/julia/extract"
+
+    job_yaml["before_script"] = [
+        "apt update && apt install -y wget",
+        "mkdir -p \$JULIA_DONWLOAD",
+        "mkdir -p \$JULIA_EXTRACT",
+        "if [[ \$CI_RUNNER_EXECUTABLE_ARCH == \"linux/arm64\" ]]; then
+  wget https://julialangnightlies-s3.julialang.org/bin/linux/aarch64/julia-latest-linux-aarch64.tar.gz -O \$JULIA_DONWLOAD/julia-nightly.tar.gz
+elif [[ \$CI_RUNNER_EXECUTABLE_ARCH == \"linux/amd64\" ]]; then
+  wget https://julialangnightlies-s3.julialang.org/bin/linux/x86_64/julia-latest-linux-x86_64.tar.gz -O \$JULIA_DONWLOAD/julia-nightly.tar.gz
+else
+  echo \"unknown runner architecture -> \$CI_RUNNER_EXECUTABLE_ARCH\"
+  exit 1
+fi",
+        "tar -xf \$JULIA_DONWLOAD/julia-nightly.tar.gz -C \$JULIA_EXTRACT",
+        # we need to search for the julia base folder name, because the second part of the name is the git commit hash
+        # e.g. julia-b0c6781676f
+        "JULIA_EXTRACT_FOLDER=\${JULIA_EXTRACT}/\$(ls \$JULIA_EXTRACT | grep -m1 julia)",
+        # copy everything to /usr to make julia public available
+        # mv is not possible, because it cannot merge folder
+        "cp -r \$JULIA_EXTRACT_FOLDER/* /usr",
+    ]
+
+    job_yaml["image"] = "debian:bookworm-slim"
```

Comment:
I think the image name `debian:bookworm-slim` should probably be a constant somewhere so it can easily be changed. It is also used in at least two places currently.